src/main/java/com/google/sps/data/AvailabilityTimeSlot.java

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.auto.value.AutoValue;
+
+/**
+ * An AvailabilityTimeSlot represents a 15 minute chunk of time that has a date and time that are
+ * expressed as readable date and time fields as well as a more all-encompassing utcEncoding. It
+ * also has a boolean that tells whether or not the time slot has been selected.
+ */
+@AutoValue
+public abstract class AvailabilityTimeSlot {
```

Comment:
This probably could be submitted as its own PR without much fanfare

(just an idea for how you could split future large PRs)

Reply:
Yeah, this PR became so huge because it started with a brainstorm/experiment to see how different files needed to interact and what the flow should be. Thus several files were effected simultaneously.